### PR TITLE
Add acceptNotification parameter to RefreshIndicator and OverscrollIn…

### PR DESCRIPTION
**Pull request by robertoscaramuzzi**
…dicator

I added a test for RefreshIndicator, but I need some guidance in adding a test for OverscrollIndicator.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scroll_notification.dart [hunk 1]
```diff
@@ -282,3 +282,13 @@ class UserScrollNotification extends ScrollNotification {
...
 }
+
+/// A predicate for [ScrollNotification], used to customize widgets that
+/// listen to notifications from their children.
+typedef bool ScrollNotificationPredicate(ScrollNotification notification);
+
+/// A [ScrollNotificationPredicate] that checks whether notification.depth == 0,
+/// which means that the notification diid not bubble through any intervening
+/// scrolling widgets.
+bool defaultScrollNotificationPredicate(ScrollNotification notification) =>
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_notification.dart]: nit: avoid => if it involves line breaks

File: packages/flutter/lib/src/material/refresh_indicator.dart [hunk 2]
```diff
@@ -112,6 +113,13 @@ class RefreshIndicator extends StatefulWidget {
   /// The progress indicator's background color. The current theme's
   /// [ThemeData.canvasColor] by default.
   final Color backgroundColor;
+  
+  /// A predicate that specifies whether a [ScrollNotification] should be
+  /// handled by this widget.
+  ///
+  /// By default, checks whether notification.depth == 0. Set it to something
```
Comment (Hixie) [inline packages/flutter/lib/src/material/refresh_indicator.dart]: put backticks around the expression for better styling

File: packages/flutter/lib/src/material/refresh_indicator.dart [hunk 3]
```diff
@@ -86,7 +86,8 @@ class RefreshIndicator extends StatefulWidget {
     this.displacement: 40.0,
     @required this.onRefresh,
     this.color,
-    this.backgroundColor
+    this.backgroundColor,
+    this.acceptNotification: defaultScrollNotificationPredicate,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/refresh_indicator.dart]: i assume you don't want this one to be null either?

Comment (Hixie) [reply packages/flutter/lib/src/material/refresh_indicator.dart]: (update the docs to be in a similar style to the other class, listing arguments that musn't be null)

Comment (Hixie) [reply packages/flutter/lib/src/material/refresh_indicator.dart]: (and assert accordingly)

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 4]
```diff
@@ -37,19 +37,21 @@ class GlowingOverscrollIndicator extends StatefulWidget {
   /// widget must contain a widget that generates a [ScrollNotification], such
   /// as a [ListView] or a [GridView].
   ///
-  /// The [showLeading], [showTrailing], [axisDirection], and [color] arguments
-  /// must not be null.
+  /// The [showLeading], [showTrailing], [axisDirection], [color] and
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart]: comma after `[color]`

File: packages/flutter/test/material/refresh_indicator_test.dart [hunk 5]
```diff
@@ -46,6 +46,48 @@ void main() {
...
+    expect(refreshCalled, false); 
+    
+
+    await tester.fling(find.text('A'), const Offset(0.0, 300.0), 1000.0); // vertical fling
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the scroll animation
+    await tester.pump(const Duration(seconds: 1)); // finish the indicator settle animation
+    await tester.pump(const Duration(seconds: 1)); // finish the indicator hide animation
+    expect(refreshCalled, true); 
+  });
```
Comment (Hixie) [inline packages/flutter/test/material/refresh_indicator_test.dart]: need a test for the overscroll indicator too

**Comment by Hixie** [comment]
https://github.com/flutter/flutter/blob/master/packages/flutter/test/widgets/overscroll_indicator_test.dart has examples of tests for OverscrollIndicator

**Comment by Hixie** [comment]
I think we need a better name than `acceptNotification`. That sounds like it's a notification of some sort. It took me a while to figure out what it was (even reading the docs wasn't immediately illuminating, so maybe the docs should be fleshed out too...)

Maybe the property name should be something like "onNotification" or "onScroll" (would be more consistent with onRefresh, which also returns an API-relevant value)? Or something a bit more on the nose like "notificationPredicate"?

For the docs, probably avoid using the word "predicate" in the first paragraph, a lot of our developers don't speak English as their first language. This is especially the case if you do put predicate in the property name.

File: packages/flutter/lib/src/widgets/scroll_notification.dart (hunk 1 above)
Comment (robertoscaramuzzi) [reply packages/flutter/lib/src/widgets/scroll_notification.dart]: Done.

File: packages/flutter/lib/src/material/refresh_indicator.dart (hunk 2 above)
Comment (robertoscaramuzzi) [reply packages/flutter/lib/src/material/refresh_indicator.dart]: Done

File: packages/flutter/lib/src/material/refresh_indicator.dart (hunk 3 above)
Comment (robertoscaramuzzi) [reply packages/flutter/lib/src/material/refresh_indicator.dart]: Done

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart (hunk 4 above)
Comment (robertoscaramuzzi) [reply packages/flutter/lib/src/widgets/overscroll_indicator.dart]: Done

File: packages/flutter/test/material/refresh_indicator_test.dart (hunk 5 above)
Comment (robertoscaramuzzi) [reply packages/flutter/test/material/refresh_indicator_test.dart]: Done

**Comment by robertoscaramuzzi** [comment]
Changed the parameter name to `notificationPredicate`; onNotification and onScroll sound too much like event handlers to me.

**Comment by robertoscaramuzzi** [comment]
Can you PTAL?

**Comment by Hixie** [comment]
LGTM

**Pull request merged**